### PR TITLE
reverse axis tests + xlim() ylim() lim() function updates

### PR DESCRIPTION
**Pull request by lindsayplatt**
fixing #352 and #356

### CODE REVIEW

File: R/gsplot.R [hunk 1]
```diff
@@ -16,8 +16,9 @@ pkg.env <- new.env()
       list(package='gsplot', def.funs=gsplot::callouts.default),
     'error_bar' =
       list(package='gsplot', def.funs=gsplot::error_bar.default),
+    
     "par" = c(),"axis" = c(), "abline" = c(), "legend" = c(), 
-    "title" = c(), "text" = c(), "mtext" = c(), "grid" = c(),
```
Comment (jordansread) [inline R/gsplot.R:20]: did this need to be removed?

File: R/access-gsplot.R [hunk 2]
```diff
@@ -113,31 +117,55 @@ lim <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,
...
-      lims <- lim(object, axis=as.axis(side))
-      sides <- as.side(names(lims)[sapply(lims, function(x) !any(is.na(x)))])
-      closest.side <- sides[which.min(abs(side-sides))][1]
-      if (is.null(closest.side)){
-        lims <- NULL
-      } else {
-        lims <- lims[[as.side_name(closest.side)]]  
+  names(lims) <- all.side.names
+  
+  if(set.undefined){
```
Comment (jordansread) [inline R/access-gsplot.R:129]: do you need to update the tests for limits or do we have decent coverage of these new changes?

File: R/access-gsplot.R [hunk 3]
```diff
@@ -113,31 +117,55 @@ lim <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,
...
-      closest.side <- sides[which.min(abs(side-sides))][1]
-      if (is.null(closest.side)){
-        lims <- NULL
-      } else {
-        lims <- lims[[as.side_name(closest.side)]]  
+  names(lims) <- all.side.names
+  
+  if(set.undefined){  
+    # get names of all sides on the same axis (x or y) that are not completely NA
+    which.undef <- sapply(lims, function(x) all(is.na(x)))
```
Comment (jordansread) [inline R/access-gsplot.R:131]: :+1:

File: R/gsplot.R (hunk 1 above)
Comment (lindsayplatt) [inline R/gsplot.R:20]: yes because it is defined on line 9-10 above

File: R/access-gsplot.R (hunk 2 above)
Comment (lindsayplatt) [inline R/access-gsplot.R:129]: pretty good coverage between test-access_gsplot and tests-axis

File: R/access-gsplot.R [hunk 4]
```diff
@@ -113,31 +117,55 @@ lim <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,
...
   }
-    
+  
+  if (!is.null(lims) && !is.null(side) && length(side)==1){ ## move this to the end
+    lims <- lims[[side.names]]
+  } else {
+    lims <- lims[side.names]
+  }
+  
+  if (is.null(lims)){
```
Comment (jordansread) [inline R/access-gsplot.R:166]: does this only apply to a single limit? 

``` r
gs <- gsplot() %>% points(1,2)
lim(gs, c(3,4))
```

File: R/access-gsplot.R (hunk 2 above)
Comment (jordansread) [inline R/access-gsplot.R:129]: yep, and I think your test addition will cover this missing spot
https://coveralls.io/builds/6824587/source?filename=R%2Faccess-gsplot.R#L119

File: R/access-gsplot.R [hunk 5]
```diff
@@ -97,11 +97,15 @@ ylim.gsplot <- function(object, side=NULL, set.undefined=TRUE){
...
-  if (!is.null(side))
+#' @export
+lim.gsplot <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,1)){
+  all.side.names <- names(sides(object))
+  side.names <- all.side.names
+  if (!is.null(side)) {
     side.names <- as.side_name(side)
-  else {
+    side.axis <- as.axis(side)
+    if(!is.null(axis) && side.axis != axis){ stop(paste("side", side, "does not have", axis, "limits"))}
```
Comment (lindsayplatt) [inline R/access-gsplot.R]: `!axis %in% side.axis`

**Comment by lindsayplatt** [comment]
When side is not specified, then it is not showing up in the object and therefore not seen as "undefined"

``` r
gs <- gsplot() %>% 
    points(1:10,11:20, side=c(2,3))

ylim(gs,4)
[1] 0 1
xlim(gs,1)
[1] 0 1
```

**Comment by jordansread** [comment]
👍

**Comment by lindsayplatt** [comment]
Still need to fix stuff based on my last comment, so that will be in a new PR :)

**Comment by jordansread** [comment]
oh, didn't realize this was unfinished. sorry

**Pull request merged**